src/legacy/java/net/neoforged/moddevgradle/legacy/RemapJarTask.java

```diff
@@ -0,0 +1,46 @@
+package net.neoforged.moddevgradle.legacy;
+
+import org.gradle.api.file.ConfigurableFileCollection;
+import org.gradle.api.file.RegularFileProperty;
+import org.gradle.api.tasks.InputFile;
+import org.gradle.api.tasks.InputFiles;
+import org.gradle.api.tasks.Nested;
+import org.gradle.api.tasks.Optional;
+import org.gradle.api.tasks.TaskAction;
+import org.gradle.jvm.tasks.Jar;
+import org.gradle.process.ExecOperations;
+
+import javax.inject.Inject;
+import java.io.IOException;
+
+/**
+ * Task used to remap a jar using AutoRenamingTool.
+ */
+public abstract class RemapJarTask extends Jar {
```

Comment:
The reason I made it inherit was not only because that handles the file name, but because it's needed for gradle to figure out its classifier as an artifact without manual input